src/messages/MessageBuilder.cpp

```diff
@@ -420,13 +421,21 @@ EmotePtr makeSharedChatBadge(const QString &sourceName,
 EmotePtr parseEmote(TwitchChannel *twitchChannel, const EmoteName &name)
 {
     // Emote order:
+    //  - Channel emotes
     //  - FrankerFaceZ Channel
     //  - BetterTTV Channel
     //  - 7TV Channel
+    //  - Global emotes
     //  - FrankerFaceZ Global
     //  - BetterTTV Global
     //  - 7TV Global
```

Comment:
I'd rather user-specific emotes don't take precedence over something a streamer controls

Reply:
We could resolve these at the end instead of at the top then, since this will not change the emote order

Reply:
How plugins play into this is mostly orthogonal, because they will use the priorities in `EmoteProvider`. There, we can make sure they set at least priority X.

My plan is:

- This PR
- Migration of FFZ, BTTV, 7TV
- Some plugin interface to _get_ these emotes from existing providers (e.g. search for emote by name)
- A plugin interface to create new emote providers

Reply:
That sounds reasonable, but I don't understand this part:

> We could, but then the priorities would be wrong while providers are being converted.